src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -147,21 +147,27 @@ export class ExtensionsListView extends ViewletPanel {
 		}
 
 		const successCallback = model => {
+			if (token && token.isCancellationRequested) {
+				return null;
+			}
 			this.setModel(model);
 			return model;
 		};
 		const errorCallback = e => {
 			console.warn('Error querying extensions gallery', e);
+			if (token && token.isCancellationRequested) {
+				return null;
+			}
 			const model = new PagedModel([]);
 			this.setModel(model, true);
 			return model;
 		};
 
 		if (ExtensionsListView.isInstalledExtensionsQuery(query) || /@builtin/.test(query)) {
-			return await this.queryLocal(parsedQuery, options).then(successCallback).catch(errorCallback);
+			return this.queryLocal(parsedQuery, options).then(successCallback).catch(errorCallback);
```

Comment:
Why don't you pass the cancellation token to the APIs those are being called. For example, if you pass the token to `queryGallery` it will also benefit in cancelling the request it made.

Reply:
done